core/src/main/java/tech/pegasys/eth2signer/core/Runner.java

```diff
@@ -134,24 +135,32 @@ public void run() {
           factory.createBlsSignerProvider(config.getKeyConfigPath(), config.getKeyCacheLimit());
       blsSignerProvider.cacheAllSigners();
 
-      final DirectoryBackedArtifactSignerProvider secpSignerProvider =
-          factory.createSecpSignerProvider(config.getKeyConfigPath(), config.getKeyCacheLimit());
-      secpSignerProvider.cacheAllSigners();
+      final DirectoryBackedArtifactSignerProvider ethSecpSignerProvider =
+          factory.createEthSecpSignerProvider(config.getKeyConfigPath(), config.getKeyCacheLimit());
+      ethSecpSignerProvider.cacheAllSigners();
 
-      final PublicKeys publicKeys = new PublicKeys(blsSignerProvider, secpSignerProvider);
+      final DirectoryBackedArtifactSignerProvider fcSecpSignerProvider =
+          factory.createFilecoinSecpSignerProvider(
+              config.getKeyConfigPath(), config.getKeyCacheLimit(), config.getFilecoinNetwork());
```

Comment:
going to need to remove KeyCachelimit

Reply:
and wondering if the getFileCoinNetwork should go into the factory's constructor - rather than on this API